desc/plotting.py

```diff
@@ -3843,3 +3845,144 @@ def plot_logo(save_path=None, **kwargs):
         fig.savefig(save_path, facecolor=fig.get_facecolor(), edgecolor="none")
 
     return fig, ax
+
+
+def plot_adiabatic_invariant(
+    eq, rhos=None, alphas=None, num_pitch=None, pitch_idx=-1, mode="single-surface"
+):
+    """Plot the second adiabatic invariant J_|| (parallel action).
+
+    This function visualizes the parallel action invariant J_|| in either of two modes:
+    1. On a single flux surface as a function of fieldline label α and inverse
+       pitch angle 1/λ
+    2. As contours of J_|| for a fixed pitch angle across multiple flux surfaces
+
+    Parameters
+    ----------
+    eq : object
+        Equilibrium object containing magnetic field information
+    rhos : array_like or float, optional
+        Flux surface radii. If float, plots single surface.
+        Default: np.linspace(0.1, 1, 10)
+    alphas : array_like, optional
+        Fieldline label values (toroidal angle).
+        Default: np.linspace(0, 2π, 32, endpoint=True)
+    num_pitch : int, optional
+        Number of pitch angle values for bounce integral calculation.
+        Default: 16
+    pitch_idx : int, optional
```

Comment:
rather than an index can the user specify a particular value of pitch angle to plot?

Reply:
That could be confusing for the user. They would have to find the |B| on that surface, invert it to get the pitch angle and then pick a  value. What we can do is give them a variable t (like Caoxiang's paper) that goes from 0-> 1 linearly covering values from the lowest pitch to the highest